base/essentials.jl

```diff
@@ -668,7 +668,7 @@ Change the type-interpretation of the binary data in the isbits value `x`
 to that of the isbits type `Out`.
 The size (ignoring padding) of `Out` has to be the same as that of the type of `x`.
 For example, `reinterpret(Float32, UInt32(7))` interprets the 4 bytes corresponding to `UInt32(7)` as a
-[`Float32`](@ref).
+[`Float32`](@ref). This operation is guaranteed to be reversible.
```

Comment:
Maybe just state explicitly what this means? E.g. like this
```suggestion
[`Float32`](@ref). Note that `reinterpret(In, reinterpret(Out, x)) === x`
```

Reply:
Is this true? It sounds like it is trying to imply that it sanitized UB and inconsistent behavior (such as fast math flags), but that might not be true of this operator

Reply:
It's likely I'm missing some nuance, but I don't see the opportunity for something undefined here given that it is strictly copying X non-padding bytes, to another container that has X non-padding bytes. How would fast math etc. potentially interfere? And if so, does this doc need to mention what happens, or would it be sufficient for the fast math flag docs to make clear that operations like this can break?

Reply:
fastmath I believe already breaks the contract that `x === x`, so I guess that is not actually overly relevant here, so I think actually it does seem the stronger claim here is justified (within the context of Julia's intrinsics already not permitting UB to be exposed)